src/components/modals.tsx

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+
+import {
+  Button,
+  ButtonVariant,
+  CloseModalCrossButton,
+} from "./buttons/buttons";
+import { ModalOverlay } from "./modal-overlay";
+
+enum ModalVariant {
+  MARK_AS_PRIMARY = "MARK_AS_PRIMARY",
+  DELETION = "DELETION",
```

Comment:
I find the naming a bit to specific, maybe something like: `ACTION` and `DESTRUCTIVE_ACTION` (or like so)?